lib/ui/src/components/preview/libs/getTools.tsx

```diff
@@ -0,0 +1,190 @@
+import React, { Fragment } from 'react';
+import window from 'global';
+import copy from 'copy-to-clipboard';
+import memoize from 'memoizerific';
+import { types } from '@storybook/addons';
+import { styled } from '@storybook/theming';
+import { Icons, IconButton, TabButton, TabBar, Separator } from '@storybook/components';
+import { ZoomConsumer, Zoom } from '../zoom';
+import { BackgroundConsumer } from '../background';
+import { stringifyQueryParams } from './stringifyQueryParams';
+import * as S from '../components';
+
+const DesktopOnly = styled.span({
+  // Hides full screen icon at mobile breakpoint defined in app.js
+  '@media (max-width: 599px)': {
+    display: 'none',
+  },
+});
+
+/**
+ * FOUNDATION TYPES
+ * TODO: below is not strictly typed yet
+ */
+export interface StorybookUIElement {
+  id: string;
+  title?: string;
+  route?: (props: any) => string;
+  active?: (props: any) => boolean;
+  match?: (props: any) => boolean;
+  render: (props: any) => React.ReactNode;
+}
+
+/**
+ * HELPERS
+ */
+export const getElementList = memoize(10)((getFn, type, base) =>
+  base.concat(Object.values(getFn(type)))
+);
+
+type matchViewMode = (mode: string) => (arg: { viewMode: string }) => boolean;
+const matchOnViewMode: matchViewMode = mode => ({ viewMode }) => viewMode === mode;
+
+/**
+ * Components extraction
+ * TODO: components below would get typed and be pull out from the current file eventually.
+ */
+const TabsTool = ({ panels, storyId, viewMode, path, location }) => (
+  <>
+    <TabBar key="tabs" {...{ scroll: false } as any}>
```

Comment:
Need to escape it to `any`, `TabBar` don't have `scroll` defined in its interface.  I think there are possibly have other similar typing defeats. I don't want to fix here there across packages (since it is not run-time bug). Will need to revisit to fix them altogether.

Reply:
I think that prop might have been removed? hence why it's not in the interface?

Reply:
Will need deeper investigation, but so far I want to start to refract it instead of changing its behaviour.